packages/payment/src/models/payment-collection.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  BeforeCreate,
+  Cascade,
+  Collection,
+  Entity,
+  Enum,
+  Filter,
+  ManyToMany,
+  OneToMany,
+  OnInit,
+  OptionalProps,
+  PrimaryKey,
+  Property,
+} from "@mikro-orm/core"
+import { DAL } from "@medusajs/types"
+
+import { DALUtils, generateEntityId } from "@medusajs/utils"
+import PaymentProvider from "./payment-provider"
+import PaymentSession from "./payment-session"
+import Payment from "./payment"
+
+/**
+ * @enum
+ *
+ * The payment collection's status.
+ */
+export enum PaymentCollectionStatus {
+  /**
+   * The payment collection isn't paid.
+   */
+  NOT_PAID = "not_paid",
+  /**
+   * The payment collection is awaiting payment.
+   */
+  AWAITING = "awaiting",
+  /**
+   * The payment collection is authorized.
+   */
+  AUTHORIZED = "authorized",
+  /**
+   * Some of the payments in the payment collection are authorized.
+   */
+  PARTIALLY_AUTHORIZED = "partially_authorized",
+  /**
+   * The payment collection is canceled.
+   */
+  CANCELED = "canceled",
+}
+
+type OptionalPaymentCollectionProps =
+  | "authorized_amount"
+  | "refunded_amount"
+  | "region_id"
+  | "completed_at"
+  | DAL.SoftDeletableEntityDateColumns
+
+@Entity({ tableName: "payment_collection" })
+@Filter(DALUtils.mikroOrmSoftDeletableFilterOptions)
+export default class PaymentCollection {
+  [OptionalProps]?: OptionalPaymentCollectionProps
+
+  @PrimaryKey({ columnType: "text" })
+  id: string
+
+  @Property({ columnType: "text" })
+  currency_code: string
+
+  @Property({
+    columnType: "numeric",
+    serializer: Number,
+  })
+  amount: number
+
+  @Property({
+    columnType: "numeric",
+    nullable: true,
+    serializer: Number,
```

Comment:
todo: maybe we should use `optionalNumericSerializer` otherwise the serializer will have issues

Reply:
good point, fixed